pkg/agent/http_test.go

```diff
@@ -119,184 +216,22 @@ func Test_accessControl(t *testing.T) {
 
 	agt := mockAgent(t)
 	httpBackend := agt.httpBackend()
-
-	func() {
-		t.Log("federate testing begin ...")
-
-		tokenScenariosMap := map[string]map[string]samples.Scenario{
-			"noneNamespacesToken": samples.NoneNamespacesTokenFederateScenarios,
-			"someNamespacesToken": samples.SomeNamespacesTokenFederateScenarios,
-		}
-
-		for token, tokenScenarios := range tokenScenariosMap {
-			for name, tokenScenario := range tokenScenarios {
-				req := httptest.NewRequest("GET", "http://example.org/federate?"+tokenScenario.Queries.Encode(), nil)
```

Comment:
All of the logic within these for loops was moved into `ScenarioValidator.Validate()`, with switch cases depending on the type of tests we want to run. This avoid a LOT of redundant code, such as this httptest stuff.